aws/resource_aws_ses_email_identity_test.go

```diff
@@ -0,0 +1,146 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSSESEmailIdentity_basic(t *testing.T) {
+	email := fmt.Sprintf(
+		"%s@terraformtesting.com",
+		acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSESEmailIdentityDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsSESEmailIdentityConfig(email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSESEmailIdentityExists("aws_ses_email_identity.test"),
+					testAccCheckAwsSESEmailIdentityArn("aws_ses_email_identity.test", email),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSSESEmailIdentity_trailingPeriod(t *testing.T) {
+	email := fmt.Sprintf(
+		"%s@terraformtesting.com.",
+		acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSESEmailIdentityDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsSESEmailIdentityConfig(email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSESEmailIdentityExists("aws_ses_email_identity.test"),
+					testAccCheckAwsSESEmailIdentityArn("aws_ses_email_identity.test", email),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
Since the email argument is marked as ForceNew: true, an additional Test should be added to the acceptance test to check that a resource has been recreated.

Reply:
I'm not sure if I understand this one. My understanding of ForceNew is that it creates a new resource when the field changes, but the email is the ID so doesn't a change explicitly create a new resource?